op-service/testutils/random.go

```diff
@@ -275,6 +297,40 @@ func RandomBlock(rng *rand.Rand, txCount uint64) (*types.Block, []*types.Receipt
 	return RandomBlockPrependTxs(rng, int(txCount))
 }
 
+func RandomBlockPrependTxsWithTime(rng *rand.Rand, txCount int, t time.Time, ptxs ...*types.Transaction) (*types.Block, []*types.Receipt) {
```

Comment:
same here, would be great to deduplicate by having `RandomBlockPrependTxs` call this one